src/item.h

```diff
@@ -1442,7 +1445,23 @@ class item : public visitable<item>
          */
         /*@{*/
         bool has_flag( const std::string &flag ) const;
-        bool has_any_flag( const std::vector<std::string> &flags ) const;
+
+        template<typename Iter, typename T = std::decay_t<decltype( *begin( std::declval<Iter>() ) )>>
+        bool has_any_flag( const Iter &flags ) const {
+            return std::any_of( flags.begin(), flags.end(), [&]( const T & flag ) {
+                return has_flag( flag );
+            } );
+        }
```

Comment:
I'm guessing this is the template you're talking about?  I would change a couple of things:
* `Iter` is actually a container type, so it probably shouldn't be called `Iter`; maybe `Container`.
* You're using non-member `begin` in the template argument but member `begin` in the body.  You should be consistent about which you're using.  Technically non-member `begin` is slightly better, but this codebase mostly uses member `begin`, so I'd stick with that.
* You should probably use `declval<const Container&>` to match the argument type.
* `has_flag` has to take a `std::string`, so the lambda *could* just take a `std::string` too.  But it doesn't really matter either way.  This is fine too.

Reply:
Thanks for the review! 

This was not the template I asked about on discord, but, as you see, I could use your insight on this one too.

One small notice, regarding "`has_flag` has to take a `std::string`", this PR is a groundwork for having `has_flag` that takes either `std::string`, `flag_str_id`, or `flag_id` , hope now it makes more sense.